internal/venice-client-common/src/main/java/com/linkedin/venice/compression/GzipPool.java

```diff
@@ -0,0 +1,51 @@
+package com.linkedin.venice.compression;
+
+import com.linkedin.venice.utils.concurrent.CloseableThreadLocal;
+import java.io.IOException;
+import java.io.OutputStream;
+
+
+abstract class GzipPool {
+  private static class ReusableObjects implements AutoCloseable {
+    final SettableOutputStream stream = new SettableOutputStream();
+    final ReusableGzipOutputStream gzipOutputStream = new ReusableGzipOutputStream(stream);
+
+    @Override
+    public void close() throws Exception {
+      stream.close();
+    }
+  }
+
+  private static final CloseableThreadLocal<ReusableObjects> reusableObjectsThreadLocal =
+      new CloseableThreadLocal(ReusableObjects::new);
```

Comment:
Hmm... this is not actually going to close anything, actually, since we never call close on it. I think I would prefer if we didn't have any static functions here, and instead made the `GzipPool` a member variable of the `GzipCompressor`, and have that class' close function call the GzipPool's close, which itself would call close on the `reusableObjectsThreadLocal`.